app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityUnregister.java

```diff
@@ -0,0 +1,145 @@
+package com.sdp.movemeet.view.activity;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageReference;
+import com.google.firebase.storage.UploadTask;
+import com.sdp.movemeet.backend.FirebaseInteraction;
+import com.sdp.movemeet.view.home.LoginActivity;
+import com.sdp.movemeet.R;
+import com.sdp.movemeet.models.Activity;
+
+public class ActivityDescriptionActivityUnregister extends AppCompatActivity {
+
+    FirebaseAuth fAuth;
+    private Activity act;
+    private static final String TAG = "ActDescActivity";
+    StorageReference storageReference;
+    ImageView activityImage;
+    String imagePath;
+    ProgressBar progressBar;
+    Uri uri;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_description);
+
+        Intent intent = getIntent();
+
+        if (intent != null) {
+            act = (Activity) intent.getSerializableExtra("activity");
+        }
+
+        uri = intent.getData();
+        if(uri != null){
+            loadActivityHeaderPicture();
+        }
+
+        createTitleView();
+        createParticipantNumberView();
+        createDescriptionView();
+        createAddressView();
+        createSportView();
+        createDurationView();
+    }
+
+    public void goToLogin(View v) {
+        if (fAuth.getCurrentUser() == null) {
+            startActivity(new Intent(getApplicationContext(), LoginActivity.class)); // sending the user to the "Login" activity
+            finish();
+        }
+    }
+
+    private void createTitleView() {
+        // activityTitle from the activity
+        TextView activityTitle = (TextView) findViewById(R.id.activity_title_description);
+        if (act != null) activityTitle.setText(act.getTitle());
+    }
+
+    private void createParticipantNumberView() {
+        // number of participants from the activity
+        TextView numberParticipantsView = (TextView) findViewById(R.id.activity_number_description);
+        if (act != null) {
+            numberParticipantsView.setText(act.getParticipantId().size() + "/" + act.getNumberParticipant());
```

Comment:
Crashes if user is null. Might be interesting to look out as a future issue

Reply:
In this class, the user is always null

Reply:
We get the number of participants, but not the name of this participants